### ISSUE TITLE
`-load_index=some_junk_file` doesn't print anything

### ISSUE THREAD

**Issue by ptarjan** [livegrep/livegrep#256]
I was trying to see if I could load a server using a `.idx.tmp` file since `Finalizing...` was taking forever in the index building step.

And just running `-load_index=/data/livegrep.idx.tmp` (or any real file) just churns over it and then exits without saying anything. A helpful error message would be nice.

**Comment by nelhage** [comment]
I see an assertion failure if I pass it some random file:

```
[nelhage@penguin:~/code/livegrep]$ bazel-bin/src/tools/codesearch -load_index bazel-bin/src/tools/codesearch
codesearch: src/dump_load.cc:215: T* load_allocator::ptr(uint64_t) [with T = chunk_header; uint64_t = long unsigned int]: Assertion `off < map_size_' failed.
Aborted (core dumped)
```

which is not as nice as I might hope, but isn't silent. How are you building livegrep? I guess it's possible an optimized build is stripping those assertions…

**Comment by ptarjan** [comment]
Thanks. I’m using the same builds you have for livegrep.com so I’m assuming
those are production and don’t include the assertions. Maybe a good simple
task would be to make this a log message instead.

**Comment by nelhage** [comment]
Yeah, it looks like most or all of the `assert`s in https://github.com/livegrep/livegrep/blob/3bcd52d4531e61505117aa251265fc93de26d7b1/src/dump_load.cc (e.g. https://github.com/livegrep/livegrep/blob/3bcd52d4531e61505117aa251265fc93de26d7b1/src/dump_load.cc#L215 should be converted to `if (…) die()` ala 27e1f8b6b2117ea2f3778788b63a026e899e0c4e

I'd happily take that PR if you want to try it :)
